### PR TITLE
feat(pie-monorepo): add angular 12 example app

### PR DESCRIPTION
**Pull request by FayeCarter**
[added] - example angular 12 app
[added] - `pie-button` to angular 12 app

image

### CODE REVIEW

File: apps/examples/wc-angular12/package.json [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+        "@angular/cli": "~15.2.6",
+        "@angular/compiler-cli": "15.2.0",
+        "@types/jasmine": "~4.3.0",
+        "jasmine-core": "~4.5.0",
+        "karma": "~6.4.0",
+        "karma-chrome-launcher": "~3.1.0",
+        "karma-coverage": "~2.2.0",
+        "karma-jasmine": "~5.1.0",
+        "karma-jasmine-html-reporter": "~2.0.0",
+        "typescript": "~4.9.4"
```
Comment (ashleynolan) [inline apps/examples/wc-angular12/package.json:37]: Can you lock these dependencies (remove any `~) and from the dependencies above too.

Also, do I'm guessing we don't need the `package-lock.json` (as there's a `yarn.lock`)

File: apps/examples/wc-angular12/src/styles.scss [hunk 2]
```diff
@@ -0,0 +1,27 @@
+    @font-face {
```
Comment (ashleynolan) [inline apps/examples/wc-angular12/src/styles.scss:1]:
```suggestion
@font-face {
```

File: apps/examples/wc-angular12/package.json [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    "dependencies": {
+        "@angular/animations": "15.2.0",
+        "@angular/common": "15.2.0",
+        "@angular/compiler": "15.2.0",
+        "@angular/core": "15.2.0",
+        "@angular/forms": "15.2.0",
+        "@angular/platform-browser": "15.2.0",
+        "@angular/platform-browser-dynamic": "15.2.0",
+        "@angular/router": "15.2.0",
+        "@justeattakeaway/pie-button": "0.10.2",
```
Comment (ashleynolan) [inline apps/examples/wc-angular12/package.json:20]: In complete contrast to my other comment, can you add a `^` to the pie-button dependency (just as that'll mean it'll always try and use latest minor version).

File: apps/examples/wc-angular12/src/app/app.component.html [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+    <h2>Test click event</h2>
+    <div>
+        <h3>Counter</h3>
+        <div class="flex-wrapper">
+        <pie-button (click)="decrement()">decrement</pie-button>
+        <div class="padding">
+            Counter: {{ count }}
+        </div>
+        <pie-button (click)="increment()">increment</pie-button>
+        </div>
```
Comment (ashleynolan) [inline apps/examples/wc-angular12/src/app/app.component.html:11]:
```suggestion
        <div class="flex-wrapper">
            <pie-button (click)="decrement()">decrement</pie-button>
            <div class="padding">
                Counter: {{ count }}
            </div>
            <pie-button (click)="increment()">increment</pie-button>
        </div>
```

File: apps/examples/wc-angular12/src/app/app.component.html [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+    </div>
+    <h2>Test props</h2>
+    <div>
+        <h3>Click the button to switch the variant</h3>
+        <div class="flex-wrapper">
+        <pie-button variant={{variantName}} (click)="switchVariant()">
+            Switch variant
+        </pie-button>
+        <div class='padding'>Variant: <b>{{variantName}}</b></div>
+        </div>
```
Comment (ashleynolan) [inline apps/examples/wc-angular12/src/app/app.component.html:21]:
```suggestion
        <div class="flex-wrapper">
            <pie-button variant={{variantName}} (click)="switchVariant()">
                Switch variant
            </pie-button>
            <div class='padding'>Variant: <b>{{variantName}}</b></div>
        </div>
```

File: apps/examples/wc-angular12/package.json (hunk 3 above)
Comment (FayeCarter) [reply apps/examples/wc-angular12/package.json:20]: 😂 
I always set a task before i set up a PR to get rid of all the `^`s so 'Ash doesn't have to tell me to' 
I feel like I don't know who I am anymore 😁

Comment (ashleynolan) [reply apps/examples/wc-angular12/package.json]: There's always an exception to the rule – and if there's not you should make one up just to confuse everyone 😆

**Review by ashleynolan — APPROVED** [review]

File: apps/examples/wc-angular12/package.json [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+        "@angular/animations": "15.2.0",
+        "@angular/common": "15.2.0",
+        "@angular/compiler": "15.2.0",
+        "@angular/core": "15.2.0",
+        "@angular/forms": "15.2.0",
+        "@angular/platform-browser": "15.2.0",
+        "@angular/platform-browser-dynamic": "15.2.0",
+        "@angular/router": "15.2.0",
+        "@justeattakeaway/pie-button": "^0.10.2",
+        "@webcomponents/webcomponentsjs": "2.8.0",
```
Comment (raoufswe) [inline apps/examples/wc-angular12/package.json:21]: qq, why do we need this?

**Review by raoufswe — APPROVED** [review]

File: apps/examples/wc-angular12/src/app/app.module.ts [hunk 7]
```diff
@@ -0,0 +1,17 @@
...
+@NgModule({
+  declarations: [
+    AppComponent
+  ],
+  imports: [
+    BrowserModule
+  ],
+  providers: [],
+  bootstrap: [AppComponent],
+  schemas: [CUSTOM_ELEMENTS_SCHEMA],
```
Comment (fernandofranca) [inline apps/examples/wc-angular12/src/app/app.module.ts:15]: It seems that Angular will only understand how to deal with Web Components after the `CUSTOM_ELEMENTS_SCHEMA` is added. Isn't it worth of mention in the README?

File: apps/examples/wc-angular12/README.md [hunk 8]
```diff
@@ -0,0 +1,57 @@
...
+
+## Using Web Components in Angular 12 apps
+
+### Install the `@justeattakeaway/pie-button` package
+
+`yarn add @justeattakeaway/pie-button`
+
+### Import and use the component
+
+####
```
Comment (fernandofranca) [inline apps/examples/wc-angular12/README.md:38]: This might be a leftover, I think.

File: apps/examples/wc-angular12/src/app/app.module.ts (hunk 7 above)
Comment (FayeCarter) [reply apps/examples/wc-angular12/src/app/app.module.ts:15]: very good point

File: apps/examples/wc-angular12/package.json (hunk 6 above)
Comment (FayeCarter) [reply apps/examples/wc-angular12/package.json:21]: I don't think we do 🤦 
It was from a part of a tutorial I don't think I needed

**Review by fernandofranca — APPROVED** [review]

**Pull request merged**